### ISSUE TITLE
Error in building the modified driver for the existing kernel

### ISSUE THREAD

**Issue by Rokuho** [spanev/linux-80211n-csitool#1]
Hi! spanev

Thank you so much for providing a modified version for 4.15 kernel.
I have struggled several days using the original ones but still cannot find a proper Ubuntu version and kernel version. Every time I run the "make" command, there appears an error "No rule to make target".
Then, I found your branch which is so exited that the Ubuntu 16.04 with 4.15 can work.
However, when I do the "sudo make", there comes the error like follows:

sudo make -C /lib/modules/$(uname -r)/build M=$(pwd)/drivers/net/wireless/intel/iwlwifi INSTALL_MOD_DIR=updates \modules_install
make: Entering directory '/usr/src/linux-headers-4.15.0-45-generic'
  INSTALL /home/hangsong/linux-80211n-csitool/drivers/net/wireless/intel/iwlwifi/dvm/iwldvm.ko
At main.c:160:
- SSL error:02001002:system library:fopen:No such file or directory: bss_file.c:175
- SSL error:2006D080:BIO routines:BIO_new_file:no such file: bss_file.c:178
sign-file: certs/signing_key.pem: No such file or directory
  INSTALL /home/hangsong/linux-80211n-csitool/drivers/net/wireless/intel/iwlwifi/iwlwifi.ko
At main.c:160:
- SSL error:02001002:system library:fopen:No such file or directory: bss_file.c:175
- SSL error:2006D080:BIO routines:BIO_new_file:no such file: bss_file.c:178
sign-file: certs/signing_key.pem: No such file or directory
  INSTALL /home/hangsong/linux-80211n-csitool/drivers/net/wireless/intel/iwlwifi/mvm/iwlmvm.ko
At main.c:160:
- SSL error:02001002:system library:fopen:No such file or directory: bss_file.c:175
- SSL error:2006D080:BIO routines:BIO_new_file:no such file: bss_file.c:178
sign-file: certs/signing_key.pem: No such file or directory
  DEPMOD  4.15.0-45-generic
make: Leaving directory '/usr/src/linux-headers-4.15.0-45-generic'

Can you please give me some hint on how to fix this ?
Or is the 4.15.0-45-generic kernel not supported at present?

Thank you very much!

**Comment by spanev** [comment]
Hi @Rokuho,

Thanks for your interest in this version of the CSI driver.

The error you you are getting is due to the fact that the driver is not signed and thus doesn't have a SSL key. That should not be a problem for you to install the driver according to the instructions. Just ignore that messages. I have tested it and and it works for me. I also have put a little tip beneath the driver installation command to explain this, but I will try to make it more clear.

I hope that this answer would help.

Please let me know if you manage to make it work.

Regards,
Stan

**Comment by Rokuho** [comment]
Hi, @spanev 

Thank you for your swift reply.
As you said, the error has no influence on the install of the driver.
However, after all the steps are conducted, it seems that the NIC cannot connect to the WIFI signals.
The WIFI spots can be displayed, but cannot be connected....
Ops....

**Comment by spanev** [comment]
Hi @Rokuho,

I think that it happens if the access point you are trying to connect to is WPA2Enterprice protected. Can you try to connect to a WPA2Personal or a no encryption network? I think that I had such a problem in the past and this is how I managed to resolve it.

**Comment by Rokuho** [comment]
@spanev 

Yeah! That's true! I connect it to a non-encrypted wifi AP, it works! haha!
Thank you !

**Comment by zhangshaohu** [comment]
Does this tool work with injection mode?
